tool/tctl/common/scoped_command.go

```diff
@@ -52,9 +61,123 @@ func (c *ScopedCommand) Initialize(app *kingpin.Application, _ *tctlcfg.GlobalCL
 	}
 
 	c.tokens.Initialize(scoped, config)
+
+	c.status = scoped.Command("status", "Show the status of scoped resources")
 }
 
 // TryRun takes the CLI command as an argument (like "scoped tokens") and executes it.
 func (c *ScopedCommand) TryRun(ctx context.Context, cmd string, clientFunc commonclient.InitFunc) (match bool, err error) {
-	return c.tokens.TryRun(ctx, cmd, clientFunc)
+	var commandFunc func(ctx context.Context, client *authclient.Client) error
+	switch cmd {
+	case c.status.FullCommand():
+		commandFunc = c.Status
+	default:
+		// may match the token family of commands
+		return c.tokens.TryRun(ctx, cmd, clientFunc)
+	}
+
+	client, closeFn, err := clientFunc(ctx)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	defer closeFn(ctx)
+
+	return true, trace.Wrap(commandFunc(ctx, client))
+}
+
+// Status shows the status of scoped resources.
+func (c *ScopedCommand) Status(ctx context.Context, client *authclient.Client) error {
+	// TODO(fspmarshall/scopes): move the calculation of counts server-side and based on
+	// scoped access cache. Ideally we want to allow for a paginated view of scope stats
+	// with one item/row per scope.  We should also consider how to optionally include
+	// additional columns conditional on user's permissions (e.g. number of tokens if
+	// user has permission to view tokens... in that case specifically we want readnosecrets
+	// to allow seeing the token count without seeing actual tokens).
+
+	var roles []*scopedaccessv1.ScopedRole
+	var cursor string
+	for {
+		rsp, err := client.ScopedAccessServiceClient().ListScopedRoles(ctx, &scopedaccessv1.ListScopedRolesRequest{
+			PageToken: cursor,
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		roles = append(roles, rsp.Roles...)
+		cursor = rsp.NextPageToken
+		if cursor == "" {
+			break
+		}
+	}
+
+	var assignments []*scopedaccessv1.ScopedRoleAssignment
+	cursor = ""
+	for {
+		rsp, err := client.ScopedAccessServiceClient().ListScopedRoleAssignments(ctx, &scopedaccessv1.ListScopedRoleAssignmentsRequest{
```

Comment:
I don't think there's any reason these have to be serial, so we might get some more responsiveness paging both resources concurrently.